nuxeo-features/nuxeo-elasticsearch/nuxeo-elasticsearch-audit/src/main/java/org/nuxeo/elasticsearch/audit/ESAuditBackend.java

```diff
@@ -716,4 +735,97 @@ protected String getESIndexName() {
         return esa.getIndexNameForType(ElasticSearchConstants.ENTRY_TYPE);
     }
 
+    @Override
+    public void append(List<String> jsonEntries) {
+        BulkRequest bulkRequest = new BulkRequest();
+        for (String json : jsonEntries) {
+            try {
+                String entryId = new JSONObject(json).getString(LOG_ID);
```

Comment:
Possible improvement: as all you need is the id it may be faster to use the Jackson Streaming API (`ObjectMapper. OBJECT_MAPPER.getJsonFactory().createParser(json)` then call `jsonParser.nextToken()` etc. (but check for nesting by counting `JsonToken.START_OBJECT`/`JsonToken.END_OBJECT` to only read ids at the first level))